utils/src/main/kotlin/Extensions.kt

```diff
@@ -362,11 +362,13 @@ fun String.percentEncode(): String =
         .replace("%7E", "~")
 
 /**
- * Strip any user name / password off the URL represented by this [String]. Return the unmodified [String] if it does
- * not represent a URL or if it does not include a user name.
+ * Strip or replace any user name / password off the URL represented by this [String]. Return the unmodified [String] if
+ * it does not represent a URL or if it does not include a user name.
  */
-fun String.stripCredentialsFromUrl() =
-    toUri { URI(it.scheme, null, it.host, it.port, it.path, it.query, it.fragment).toString() }.getOrDefault(this)
+fun String.replaceCredentialsFromUrl(userInfo: String? = null) =
```

Comment:
As [mentioned](https://github.com/oss-review-toolkit/ort/pull/4138#discussion_r648232567), the new name should be `replaceCredentialsInUri`. So "In" instead of "From" and "Uri" instead of "Url".